prometheus_juju_backup_all_exporter/config.py

```diff
@@ -0,0 +1,42 @@
+import os
+from enum import Enum
+from logging import getLogger
+
+from pydantic import BaseModel, validator
+from yaml import safe_load
+
+logger = getLogger(__name__)
+
+DEFAULT_CONFIG = os.path.join(os.environ.get("SNAP_DATA"), "config.yaml")
+
+
+class LogLevel(str, Enum):
+    DEBUG = "DEBUG"
+    INFO = "INFO"
+    WARNING = "WARNING"
+    ERROR = "ERROR"
+    CRITICAL = "CRITICAL"
+
+
+class Config(BaseModel):
+    port: int = 10000
+    level: LogLevel = LogLevel.DEBUG
+
+    @validator("port")
+    def validate_port_range(cls, port):  # noqa: N805
+        if not 1024 <= port <= 10000:
```

Comment:
What's the reasoning behind disallowing ports below `1024`? To me, a port `80` is quiet reasonable choice for http service.

Reply:
Ports below 1024 are well known ports, we may not want to use these ports if not necessary (of course for internal use, it make not sense to restrict it). Also, there're some community practice to choose a default exporter port:
- https://prometheus.io/docs/instrumenting/writing_exporters/#port-numbers
- https://github.com/prometheus/prometheus/wiki/Default-port-allocations

Reply:
I'm not saying we should use well known ports by default, but I don't like this arbitrary limitation. Why disallow operator to use valid values?
Also, now I realized that upper limit is "10000". This is another arbitrary limit that does not make much sense to me. imo port validation should only check if the value is a number from 1 to 65535

Reply:
I am okay with this limitation. We are not writing a general function to validate network connections. Therefore, if there is common practice of exporter. (I also heard about it before). So I'm okay to apply it here. It also helps the COE to use the tool in a standard way.

Reply:
But from what I've read in  the docs, there's no "common practice" for the exporter ports. That part of the documentation only says that a [public list of exporters and associated addresses exists](https://github.com/prometheus/prometheus/wiki/Default-port-allocations) and if anyone is developing an exporter, they can "reserve" a port on that list.
Or am I misunderstanding it?

Reply:
I've updated the README.md to include snap configuration.